core/src/avm2/globals/math.rs

```diff
@@ -90,7 +91,9 @@ pub fn max<'gc>(
         let val = arg.coerce_to_number(activation)?;
         if val.is_nan() {
             return Ok(f64::NAN.into());
-        } else if val > cur_max {
+        } else if val > cur_max
+            || (val == cur_max && !val.is_sign_negative() && cur_max.is_sign_negative())
```

Comment:
That's an oddly specific condition. Why does it have to be here? Can't it be somehow generalized? (Similarly the `min` one.)